pinot-common/src/main/java/org/apache/pinot/common/function/TransformFunctionType.java

```diff
@@ -134,7 +152,14 @@ public enum TransformFunctionType {
   CLPDECODE("clpDecode"),
 
   // Regexp functions
-  REGEXP_EXTRACT("regexpExtract"),
+  REGEXP_EXTRACT("regexpExtract", "regexp_extract"),
+  REGEXPREPLACE("regexpReplace",
+      ReturnTypes.VARCHAR_2000_NULLABLE,
```

Comment:
we dont have 2000 limit on VARCHAR so there's no need to add the cap in varchar (or do we?)

Reply:
there is no `ReturnTypes.VARCHAR`, only `VARCHAR_4`or `VARCHAR_2000`
<img width="344" alt="image" src="https://github.com/apache/pinot/assets/1202120/afc15901-bcfa-4e1d-9186-9a7363362742">